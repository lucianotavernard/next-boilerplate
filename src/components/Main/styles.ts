import styled from 'styled-components'

export const Container = styled.main`
  text-align: center;

  @keyframes App-logo-spin {
    from {
      transform: rotate(0deg);
    }
    to {
      transform: rotate(360deg);
    }
  }

  .App-logo {
    height: 40vmin;

    pointer-events: none;

    @media (prefers-reduced-motion: no-preference) {
      animation: App-logo-spin infinite 20s linear;
    }
  }

  .App-header {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    min-height: 100vh;

    color: white;
    background-color: #282c34;

    font-size: calc(10px + 2vmin);
  }

  .App-link {
    color: #067df7;
  }
`
