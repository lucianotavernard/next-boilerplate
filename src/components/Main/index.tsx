import { Container } from './styles'

export function Main() {
  return (
    <Container>
      <header className="App-header">
        <img src="/logo.svg" className="App-logo" alt="logo" />

        <p>
          Edit <code>src/components/Main/index.tsx</code> and save to reload.
        </p>

        <a
          className="App-link"
          href="https://nextjs.org/"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn Next.js
        </a>
      </header>
    </Container>
  )
}
