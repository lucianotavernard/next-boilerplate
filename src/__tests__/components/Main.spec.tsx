import { render, screen } from '@testing-library/react'

import { Main } from '../../components/Main'

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('link', { name: /learn next.js/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    render(<Main />)

    expect(screen.getByRole('link', { name: /learn next.js/i })).toHaveStyle({
      color: '#067df7'
    })
  })
})
