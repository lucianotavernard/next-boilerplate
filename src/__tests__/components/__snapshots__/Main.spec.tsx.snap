// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main
  class="sc-bdnxRM iTZJjr"
>
  <header
    class="App-header"
  >
    <img
      alt="logo"
      class="App-logo"
      src="/logo.svg"
    />
    <p>
      Edit 
      <code>
        src/components/Main/index.tsx
      </code>
       and save to reload.
    </p>
    <a
      class="App-link"
      href="https://nextjs.org/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn Next.js
    </a>
  </header>
</main>
`
